### ISSUE TITLE
dashboard: Add view for PR runs

### ISSUE THREAD

**Issue by sprt** [kata-containers/kata-containers.github.io#1]

**Comment by afinn12** [comment]
For now, it displays only the last 5 PRs to account for rate limits.   
(PR #14 in our repository: https://github.com/a1icja/kata-dashboard-next/, looking to PR into original repo soon)

![Image](https://github.com/user-attachments/assets/c7f0bc26-9679-4d7c-9a2e-80ff5a6f1315)

**Comment by chris-krenz** [comment]
FYI, I think the reason those run counts are 0 is that the name of the test does not exactly match--in the PR, there’s an additional descriptor, e.g.:

kata-containers-ci-on-push / run-k8s-tests-on-ppc64le
vs.
kata-containers-ci-on-push / run-k8s-tests-on-ppc64le / run-k8s-tests (qemu, kubeadm)

So I think it technically is a different test (?), but I’ll investigate further.

![Image](https://github.com/user-attachments/assets/24e36759-3461-434f-ab3d-ab44169d9e8e)

![Image](https://github.com/user-attachments/assets/6866d638-43ce-4845-9872-f421d07fd164)

**Comment by sprt** [comment]
Good catch! The test is defined here btw: https://github.com/kata-containers/kata-containers/blob/main/.github/workflows/run-k8s-tests-on-ppc64le.yaml#L26

So the Github API returns [1], but the actual test name is [2], correct?

[1] kata-containers-ci-on-push / run-k8s-tests-on-ppc64le
[2] kata-containers-ci-on-push / run-k8s-tests-on-ppc64le / run-k8s-tests (qemu, kubeadm)

An idea: could it be that if you have a test `a / b / c`, the Github API returns `a / b` if `c` couldn't be executed for some reason? Look at how the test is defined here, maybe the `if` or the `needs` aren't satisfied?
https://github.com/kata-containers/kata-containers/blob/69509eff33f7c18c143628639f5a8998f3beab76/.github/workflows/ci.yaml#L200-L202

**Comment by chris-krenz** [comment]
To try and close the loop on this.  Yes, you're essentially correct, but a couple things to note.  

I think the original issue of some run counts being precisely 0 was the fact that we were using the list of jobs from the Nightly Runs when looking up the PR jobs.  But now we lookup all of the jobs from the PRs, rather than trying to cross-reference with the Nightly Run jobs.  So that specific discrepancy is resolved.

However, you are still correct that we do have cases where, because the run fails at an intermediate workflow, it will list a / b and then, on a separate row for runs that passed the intermediate step, a / b / c.

To my understanding, this is not really a problem, and those should still be displayed as separate rows with separate counts, as they currently are.  (I suppose we could aggregate them by incrementing the fails for every downstream workflow, but that would make it seem like the downstream workflow itself failed, but in reality it was simply never reached.). But happy to discuss.

**Comment by sprt** [comment]
Just dump PR into official repo
